### DISCUSSION TITLE
Java OpenApi format date

### DISCUSSION THREAD

**Discussion by danosavraam** [OAI/OpenAPI-Specification#2924] [Q&A] [answered]
Hello, I am using OpenApi in my Java project. In my yaml file I used type string and format date. But I am facing one issue with the year.
My request should come like this 2022-01-30 if I give wrong day or month validation is working and I am getting error message but for the year this is not happening. For example if I sent wrong year like this 20223131-01-30 I am not getting error message that year has to be 4 digits or there is no such year. How can I fix this?

Thank you.

**Comment by karenetheridge** [comment, marked as answer]
20223131 is a valid year, albeit not very likely to be the correct value in your application.

You could add an extra check like `"maxLength": 10"` or `"pattern": "^\d{4}-\d{2}-\d{2}$"` to ensure that you get four-digit years.

**Comment by handrews** [comment]
Closing as answered - if that `pattern` does not work, then it is a tooling issue and should be raised with the tooling vendor.  Note that support for `format` in validation is technically optional, so you need to consult your tooling documentation to determine if it is supported.  If not, the pattern given is the typical workaround.
